sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java

```diff
@@ -406,6 +408,26 @@ public void testCheckpointMarkSafety() throws Exception {
     runner.join();
   }
 
+  /** Test the checkpoint mark default coder, which is actually AvroCoder. */
+  @Test
+  public void testCheckpointMarkDefaultCoder() throws Exception {
+    JmsIO.JmsCheckpointMark jmsCheckpointMark = new JmsIO.JmsCheckpointMark();
+    jmsCheckpointMark.add(new ActiveMQMessage());
+    Coder coder = new JmsIO.UnboundedJmsSource(null).getCheckpointMarkCoder();
+    CoderProperties.coderSerializable(coder);
+    CoderProperties.coderDecodeEncodeEqual(coder, jmsCheckpointMark);
+  }
+
+  @Test
+  public void testCheckpointMarkSerializableCoder() throws Exception {
+    JmsIO.JmsCheckpointMark jmsCheckpointMark = new JmsIO.JmsCheckpointMark();
```

Comment:
This second test is not needed if you make `UnboundedJmsSource#getCheckpointMarkCoder` return the `SerializableCoder`. Also that is needed to get rid also of `AvroCoder` as part of this fixes.

Reply:
+1, I'm doing the change.